core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -815,6 +821,22 @@ private DefaultFunctionResolver timestamp() {
             TIMESTAMP, TIMESTAMP, TIMESTAMP));
   }
 
+  private DefaultFunctionResolver timestampadd() {
+    return define(BuiltinFunctionName.TIMESTAMPADD.getName(),
+        impl(nullMissingHandling(DateTimeFunction::exprTimestampAdd),
+            DATETIME, STRING, INTEGER, DATETIME),
+        impl(nullMissingHandling(DateTimeFunction::exprTimestampAdd),
+            DATETIME, STRING, INTEGER, TIMESTAMP),
+        implWithProperties(
+            nullMissingHandlingWithProperties(
+                (functionProperties, part, amount, time) -> exprTimestampAddForTimeType(
+                    functionProperties.getQueryStartClock(),
+                    part,
+                    amount,
+                    time)),
+            DATETIME, STRING, INTEGER, TIME));
```

Comment:
MySQL returns a STRING o_O
I think we need to keep DATETIME or TIMESTAMP as a return type.

Reply:
Oh interesting. Yeah that's weird. I'm not sure why they would return a string. I also think we should keep it as DATETIME or TIMESTAMP.